### PR TITLE
Message Passing and API Call

### LINKED ISSUE #8: [Chrome Extension] Update background script to make API calls to MLIS
Add following functionalities to background script
- Make API calls to MLIS
- Retrieve message sent from the MLIS
- Send data to popup component for display

### PR DESCRIPTION
**Pull request by junhur**
- Message passing between Content Script, Background, and Popup
- Makes HTTP POST Request to `localhost:8000/summary` from `background.ts`
- Update Popup UI with response received from API

Closes #8

### CODE REVIEW

**Comment by chorongi** [comment]
1. Is background.ts currently designed to support both page content summary & pop up style summary?

2. [app.py line 23](https://github.com/TeamCHK/temp-repo-name/pull/13/files#:~:text=response%20%3D%20PageContent(text%3D(content.text%20%2B%20%27%20%27%20%2B%20content.text))) was wondering if there was a specific reason behind setting input text as text=(content.text + ' ' + content.text) instead of text = context.text

**Comment by junhur** [comment]
> 1. Is background.ts currently designed to support both page content summary & pop up style summary?

A: Not entirely sure what you mean by page content summary. But right now, `background.ts` grabs text from `contentScript.ts` where article text would be. Then passes it to dummy MLIS and forwards the response to `popup.ts`

> 2. [app.py line 23](https://github.com/TeamCHK/temp-repo-name/pull/13/files#:~:text=response%20%3D%20PageContent(text%3D(content.text%20%2B%20%27%20%27%20%2B%20content.text))) was wondering if there was a specific reason behind setting input text as text=(content.text + ' ' + content.text) instead of text = context.text

A: app.py currently just serves as a dummy placeholder service for MLIS. returning `content.text` twice to provide an obvious distinction between what is coming from content script vs what is coming from service.

@chorongi  let me know if you need more clarification

File: extension/src/ts/background.ts [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
-            // TODO: replace delay with webpage load completion listener
-            await delay(3000);
-            chrome.tabs.sendMessage(tabId, {
-                url: tab.url,
-            })
-            console.log("Message sent from background script")
+    chrome.runtime.onMessage.addListener(
+        function(request, _, sendResponse) {
+            if (request.type === MsgType.PageContent) {
+                wretch(apiRoot + summarizationEndpoint)
```
Comment (sungho-cho) [inline extension/src/ts/background.ts:15]: `wretch` seems to be convenient!

File: extension/src/ts/contentScript.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 
-(() => {
-    chrome.runtime.onMessage.addListener((msg, sender, callback) => {
-        console.log('Message received from sender: ', sender.id, msg);
-        const articleTitle : String = getArticleTitle();
-        const displayText : String = articleTitle ? `The article title is ${articleTitle}` : "Not a CNN article page";
-        callback(displayText);
-    });
-})();
+if (window.location.href.includes('cnn.com') && window.location.href.includes('index.html')) {
```
Comment (sungho-cho) [inline extension/src/ts/contentScript.ts:11]: Might be useful to put a TODO here to note that we plan to add support for additional (or any) sites with future improvements :)

File: extension/src/ts/popup.tsx [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 
 
 const Popup = () => {
     const [content, setContent] = useState('');
 
     useEffect(() => {
-        chrome.tabs.query({currentWindow: true, active: true}, tabs => {
-            const currentTabId: number = tabs.length === 0 ? 0 : tabs[0].id!;
-            chrome.tabs.sendMessage(currentTabId, '', response => setContent(response));
+        const msg : PopupToBackgroundMsg = {
```
Comment (sungho-cho) [inline extension/src/ts/popup.tsx:10]: Code seems super clean with the change 👍

**Review by sungho-cho — APPROVED** [review]
Nice use of FastAPI, wretch, TypeScript types!
I approve with one nitpicky comment on contentScript.
Awesome job.

**Pull request merged**